src/main/kotlin/com/exactpro/th2/check1/rule/RuleFactory.kt

```diff
@@ -93,8 +93,10 @@ class RuleFactory(
                         "Either messageFilters or rootMessageFilters must be set but not both"
                     }
 
-                    val protoMessageFilters: List<RootMessageFilter> = request.rootMessageFiltersList.ifEmpty {
+                    val protoMessageFilters: List<RootMessageFilter> = if (request.rootMessageFiltersList.isEmpty()) {
                         request.messageFiltersList.map { it.toRootMessageFilter() }
+                    } else {
+                        request.rootMessageFiltersList.onEach { checkMessageType(it.messageType) }
                     }
```

Comment:
```suggestion
                    val protoMessageFilters: List<RootMessageFilter> =request.rootMessageFiltersList.ifEmpty {
                        request.messageFiltersList.map { it.toRootMessageFilter() }
                    }.onEach { checkMessageType(it.messageType) }
```

Reply:
If `rootMessageFiltersList` is empty, then `checkMessageType` will be called twice for each element

Reply:
Yep, but calling that method does not cost match for us. We can either remove the `checkMessageType` in `toRootMessageFilter` method or agree that calling `checkMessageType` twice for each filter is okay